L1Trigger/L1TMuonOverlapPhase1/src/Omtf/XMLEventWriter.cc

```diff
@@ -12,15 +12,47 @@
 #include "L1Trigger/L1TMuonOverlapPhase1/interface/Omtf/OMTFinput.h"
 #include "L1Trigger/L1TMuonOverlapPhase1/interface/Omtf/XMLEventWriter.h"
 
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include <boost/property_tree/xml_parser.hpp>
+
+#include <bitset>
+
 XMLEventWriter::XMLEventWriter(const OMTFConfiguration* aOMTFConfig, std::string fName)
-    : omtfConfig(aOMTFConfig), xmlWriter(aOMTFConfig), currentElement(nullptr), fName(fName) {
+    : omtfConfig(aOMTFConfig), fName(fName) {
   //std::string fName = "OMTF";
-  xmlWriter.initialiseXMLDocument("OMTF");
   eventNum = 0;
+
+  unsigned int version = aOMTFConfig->patternsVersion();
+  unsigned int mask16bits = 0xFFFF;
+
+  version &= mask16bits;
+
+  std::ostringstream stringStr;
+  stringStr.str("");
+  stringStr << "0x" << std::hex << std::setfill('0') << std::setw(4) << version;
+
+  tree.put("OMTF.<xmlattr>.version", stringStr.str());
 };
 
 XMLEventWriter::~XMLEventWriter() {}
 
+void XMLEventWriter::observeProcesorBegin(unsigned int iProcessor, l1t::tftype mtfType) {
+  if (eventNum > 5000)
+    return;
```

Comment:
Yes, the XMLEvenWriter is used for dumping the data for the firmware testing. 
If I remember correctly there was a problem here: the XML elements were stored in the memory, so the job was consuming more and more memory and was crashing finally without any meaningful information. To avoid this cut `eventNum > 5000` was hard coded.

But it was when we used xercesc::DOM here, later we moved to boost/property_tree for the XML creation, maybe it is smarter, we should check if this cut is still needed.